service/matching/version_graph.go

```diff
@@ -38,6 +41,21 @@ func ToBuildIdOrderingResponse(g *persistence.VersioningData, maxDepth int) *wor
 	return depthLimiter(g, maxDepth, true)
 }
 
+// HashVersioningData returns a farm.Fingerprint32 hash of the versioning data as bytes. If the data is nonexistent or
+// invalid, returns an empty slice.
+func HashVersioningData(data *persistence.VersioningData) []byte {
+	if data == nil || data.GetCurrentDefault() == nil {
+		return []byte{}
+	}
+	asBytes, err := proto.Marshal(data)
```

Comment:
```suggestion
	asBytes, err := data.Marshal()
```
technically `Marshal` is not guaranteed to be determinstic. but it seems safe enough here since it should be the same binary exchanging these. and the worst case is extra data.